flink-runtime/src/main/java/org/apache/flink/runtime/concurrent/impl/FlinkFuture.java

```diff
@@ -190,7 +191,16 @@ public R recover(Throwable failure) throws Throwable {
 					return Futures.future(new Callable<R>() {
 						@Override
 						public R call() throws Exception {
-							return future.get();
+							try {
+								return future.get();
+							} catch (ExecutionException e) {
+								// unwrap the execution exception if it's not a throwable
```

Comment:
Actually I meant unwrapping the execution exception if the cause is an `Exception`. Will correct the comment.